js/validation.js

```diff
@@ -17,3 +17,64 @@ form.addEventListener('submit', (e) => {
     error.classList.add('form__error-msg--active');
   }
 });
+
+function storageAvailable(type) {
+  let storage;
+  try {
+    const x = '__storage_test__';
+    storage = window[type];
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+    // everything except Firefox
+      e.code === 22
+          // Firefox
+          || e.code === 1014
+          // test name field too, because code might not be present
+          // everything except Firefox
+          || e.name === 'QuotaExceededError'
+          // Firefox
+          || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+          // acknowledge QuotaExceededError only if there's something already stored
+          && (storage && storage.length !== 0);
+  }
+}
+
+if (storageAvailable('localStorage')) {
+  // Save input values to local storage
+  const setFormValues = () => {
+    const formData = {
+      name: form.user_name.value,
+      email: form.user_email.value,
+      message: form.user_message.value,
+    };
+
+    localStorage.setItem('formData', JSON.stringify(formData));
+  };
+
+  form.user_name.addEventListener('change', setFormValues);
+  form.user_email.addEventListener('change', setFormValues);
+  form.user_message.addEventListener('change', setFormValues);
+
+  const { name, email, message } = JSON.parse(localStorage.getItem('formData'));
+
+  if (name !== '' || email !== '' || message !== '') {
+    form.user_name.value = name;
+    form.user_email.value = email;
+    form.user_message.value = message;
+  }
+
+  // Reset button
+  const resetButton = document.querySelector('.form-button--reset');
+
+  const resetForm = () => {
+    form.user_name.value = '';
+    form.user_email.value = '';
+    form.user_message.value = '';
+    localStorage.removeItem('formData');
+  };
```

Comment:
- [ ] A reset strategy is excellent when is required to delete all elements, but it will be great if this is done automatically, for example, if a put my name and then I just make empty an input. I invite you to work on this.
- Hint: A reset button is not necessary for this project.